### PR TITLE
workspace module

### PR DESCRIPTION
**Pull request by srikanthgudipudi**
details page in work space module

### CODE REVIEW

File: src/app/modules/portal-core/services/common/common.service.ts [hunk 1]
```diff
@@ -26,6 +31,22 @@ export class CommonService {
     return this.details;
   }
 
+  async getUserRoles() {
+    return new Promise((resolve, reject) => {
+      this.Http.get(environment.main_url + 'user-management/api/v1/userExtension/getProfile')
```
Comment (kiranharidas187) [inline src/app/modules/portal-core/services/common/common.service.ts]: constants should be kept in one place

File: src/app/modules/private/private/private.component.ts [hunk 2]
```diff
@@ -12,60 +12,152 @@ import { TranslateService } from '@ngx-translate/core';
...
+              },
+              {
+                "name": "Drafts",
+                "url": "/workspace/drafts"
+              },
+              {
+                "name": "Submitted for Review",
+                "url": "/workspace/submittedforreview"
+              },
+              {
```
Comment (kiranharidas187) [inline src/app/modules/private/private/private.component.ts]: Don't put this condition like this.  If he is a designer the push the workspace object

File: src/app/modules/workspace/create-page/create-page.component.ts [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+      "name": "Resources",
+      "tooltip": "create Resources"
+    },
+    {
+      "title": "Create Observations",
+      "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras volutpat auctor nibh. Curabitur malesuada dui uttellus tempus, vitae tristique erat ultrices. Suspendisse pulvinar nec nisi quis maximus.",
+      "action": "Create",
+      "redirect_url": "",
+      "endpoint": "",
+      "name": "Observations",
```
Comment (kiranharidas187) [inline src/app/modules/workspace/create-page/create-page.component.ts]: shouldn't it be in small letters?

File: src/app/modules/workspace/create-page/create-page.component.ts [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    }
+  ]
+  constructor(private router: Router, private communityService: CommunityService,
+    private commonService: CommonService) { }
+
+  ngOnInit() {
+  }
+
+ 
+  create(assesmenttype) {
```
Comment (kiranharidas187) [inline src/app/modules/workspace/create-page/create-page.component.ts]: the variables should be camelCase

File: src/app/modules/workspace/details/details.component.ts [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+      this.updateMetaData(this.form.form.value);
+    } else {
+      this.form.validateAllFormFields(this.form.form);
+    }
+  }
+
+  // To update the form
+  updateMetaData(data){
+    this.spin = true;
+    let obj = this.form.fields[4]['options'].find(o => o.value === data.entityType);
```
Comment (kiranharidas187) [inline src/app/modules/workspace/details/details.component.ts:47]: What is this for?

File: src/app/modules/portal-core/services/common/common.service.ts (hunk 1 above)
Comment (srikanthgudipudi) [reply src/app/modules/portal-core/services/common/common.service.ts]: Modified

File: src/app/modules/private/private/private.component.ts (hunk 2 above)
Comment (srikanthgudipudi) [reply src/app/modules/private/private/private.component.ts]: changed

File: src/app/modules/workspace/create-page/create-page.component.ts (hunk 4 above)
Comment (srikanthgudipudi) [reply src/app/modules/workspace/create-page/create-page.component.ts]: changed

File: src/app/modules/workspace/details/details.component.ts (hunk 5 above)
Comment (srikanthgudipudi) [reply src/app/modules/workspace/details/details.component.ts:47]: currently i am getting only value from the form, In order to send the selected object , i am filtering the data

File: src/app/modules/workspace/create-page/create-page.component.ts (hunk 3 above)
Comment (srikanthgudipudi) [reply src/app/modules/workspace/create-page/create-page.component.ts]: modified

**Pull request merged**